### PR TITLE
1475: Add unconditional exact CI for proportion differences

### LINKED ISSUE #1475: [Feature Request]: Add unconditional exact confidence-intervals for proportion differences
### Feature description

This is described in the SAS manual [here](https://documentation.sas.com/doc/en/pgmsascdc/v_074/procstat/procstat_freq_details54.htm#procstat.freq.freqrdiffcl) under "Exact Unconditional Confidence Limits":

- In particular we need to reproduce the RISKDIFF(METHOD=NOSCORE) option in the EXACT statement, where PROC FREQ computes the confidence limits by inverting two separate one-sided exact tests that are based on the unstandardized risk difference
- Method details - here we need to look at the "tail method" mentioned and use the unstandardized risk difference:

  Image

- Maybe https://github.com/cran/Exact/tree/master/R can be a good reference / starting point but we should not directly use that because there are no tests at all

- Main reference is Santner, T. J., and Snell, M. K. (1980). “Small-Sample Confidence Intervals for p 1 minus p 2 and p 1 slash p 2 in 2 times 2 Contingency Tables.” Journal of the American Statistical Association 75:386–394.

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [x] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [x] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by danielinteractive**
closes #1475

I did not commit a full roxygen run because my installed 8.0.0 version seems to change a lot of unrelated stuff.

### CODE REVIEW

**Comment by Melkiades** [comment]
@danielinteractive lets merge first #1477 so we get rid of the roxygen update

File: inst/REFERENCES.bib [hunk 1]
```diff
@@ -136,3 +136,16 @@ @article{MiettinenNurminen1985
...
+author = {Thomas J. Santner and Mark K. Snell},
+title = {Small-Sample Confidence Intervals for p1 - p2 and p1/p2 in 2 x 2 Contingency Tables},
+journal = {Journal of the American Statistical Association},
+volume = {75},
+number = {370},
+pages = {386--394},
+year = {1980},
+publisher = {Taylor \& Francis},
+doi = {10.1080/01621459.1980.10477482},
+URL = {https://doi.org/10.1080/01621459.1980.10477482}}
```
Comment (Melkiades) [inline inst/REFERENCES.bib]: there should be a tab space as for other citations

File: NEWS.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # tern 0.9.10.9006
 
+### Enhancements
+
+* Added `uncond_exact_diff` method to `estimate_proportion_diff()` for the unconditional exact confidence interval for the difference in proportions by inverting one-sided tail tests over a nuisance parameter.
```
Comment (Melkiades) [inline NEWS.md]:
```suggestion
* Added `uncond_exact_diff` method to `estimate_proportion_diff()` for the unconditional exact confidence interval for the difference in proportions by inverting one-sided tail tests over a nuisance parameter.
```

Comment (Melkiades) [reply NEWS.md]: very minor issue ahah

File: inst/WORDLIST [hunk 3]
```diff
@@ -37,14 +37,22 @@ Satterthwaite
...
 funder
+ge
+hline
+infimum
 layouting
+le
 pharmacokinetic
 pre
+qquad
 responder
```
Comment (Melkiades) [inline inst/WORDLIST:57]: (cdot, frac, ge, hline, le, qquad, infimum) look like they might be LaTeX fragments leaking from math expressions rather than actual words that need whitelisting. Worth double-checking if the spell checker is picking those up from \frac{}, \cdot, etc.

**Review by Melkiades — COMMENTED** [review]
Hey Daniel! this is a great addition :) — the Santner & Snell tail method is exactly the right approach for unconditional exact CIs, and the implementation reads very cleanly.

Added a couple of comments

File: tests/testthat/test-prop_diff.R [hunk 4]
```diff
@@ -304,6 +304,128 @@ testthat::test_that("prop_diff_strat_nc output matches equivalent SAS function o
...
+  boundary <- h_find_ci_bound_uniroot(
+    p_value_function = always_large,
+    cutoff = 0.25,
+    direction = "increasing",
+    interval = c(-1, 1)
+  )
+
+  expect_equal(boundary, -1)
+})
+
```
Comment (Melkiades) [inline tests/testthat/test-prop_diff.R:457]: tests that may be useful:

- Edge cases like all successes, all failures, or 0 events in one group
- The stratification guard (the stop() when strata are provided)

File: R/prop_diff.R [hunk 5]
```diff
@@ -896,3 +906,225 @@ prop_diff_strat_nc <- function(rsp,
...
+  n21_obs <- tbl[1, 1]
+  n11_obs <- tbl[2, 1]
+  diff_est <- n11_obs / n1 - n21_obs / n2
+
+  # Step 1: Enumerate all tables in A with fixed row margins
+  # n1 and n2.
+  tables <- expand.grid(
+    n11 = 0:n1,
+    n21 = 0:n2
+  )
```
Comment (Melkiades) [inline R/prop_diff.R:1092]: This could explode... Maybe worth adding a note in the docs about practical limits, or a soft check?

File: inst/WORDLIST (hunk 3 above)
Comment (danielinteractive) [reply inst/WORDLIST:57]: Yes, exactly. Unfortunately though this would need to be changed at the checks inherited from https://github.com/insightsengineering/r.pkg.template/blob/main/.github/workflows/spelling.yaml or in the `spelling` package directly, see also https://github.com/ropensci/spelling/issues/56

File: tests/testthat/test-prop_diff.R (hunk 4 above)
Comment (danielinteractive) [reply tests/testthat/test-prop_diff.R:457]: thanks, done now

File: R/prop_diff.R (hunk 5 above)
Comment (danielinteractive) [reply R/prop_diff.R:1092]: thanks, added warning

**Comment by llrs-roche** [comment]
For roxygen2 at 8.0.0 there is a different PR: https://github.com/insightsengineering/tern/pull/1477. Not sure which one should go first

**Comment by Melkiades** [comment]
> For roxygen2 at 8.0.0 there is a different PR: #1477. Not sure which one should go first

merged the roxygen2 PR

**Review by Melkiades — APPROVED** [review]
Hey Daniel, this is looking really good after the updates! The tests are thorough — I especially like the SAS reference values and the edge cases (all successes, all failures, zero observations).

Thank you so much for the update. Ready to go for me
